src/engine/controls/bpmcontrol.cpp

```diff
@@ -984,39 +993,41 @@ void BpmControl::trackBeatsUpdated(mixxx::BeatsPointer pBeats) {
         kLogger.trace() << getGroup() << "BpmControl::trackBeatsUpdated"
                         << (pBeats ? pBeats->getGlobalBpm() : mixxx::Bpm());
     }
-    m_pBeats = pBeats;
```

Comment:
I don't understand why removing Beats' reference to Track requires this change to BpmControl.

Reply:
`getSampleRate` is made private in `Beats` class, so it is better to retrieve sample rate from the track itself.